### ISSUE TITLE
Spacy NER: Training existing model on new entities reducing the accuracy.

### ISSUE THREAD

**Issue by AbhishekKargawal** [explosion/spaCy#3889]
I'm using Spacy NER to get organisation names from the text.
Example: 
Text: **AG Barr is recalling 750ml glass bottles due to a manufacturing fault which may cause bottle caps to pop off unexpectedly**
Entities I want:  AG Barr: ORG

Using "en_core_web_lg" as base model.
Accuracy using the base model is: 76%

From here I have filtered the incorrectly predicted sentence and created a test example of them to train new entity to improve overall accuracy. I have not created a blank model, using the "en_core_web_lg" as a base model.

After training a new entity using the new training sentence, my overall model accuracy got reduced to 60%.

**Comment by BreakBB** [comment]
Could you include the code and examples you're using?

**Comment by AbhishekKargawal** [comment]
Sentence:  **Apna  distribution recalls ‘hey I’m all yours’ Jelly Sweets because of a choking hazard.**
For the above sentence:  Spacy NER giving me output as **[('’m',   'ORG')]**
Expected output: **[('Apna  distribution',   'ORG')]**
So I thought of adding some small sentences which contain 'Apna distribution' and  I'll add a new label for this.
Examples I've added for 'Apna distribution':
![image](https://user-images.githubusercontent.com/19729714/60426059-c074a300-9beb-11e9-9092-e60db4a96960.png)
Using code I've converted these examples into required format:
**("Apna distribution could, for example, ignore all the blogs that don't have a predefined number of daily visitors.", {'entities': [(0, 16, 'P_ORG')]})**


![image](https://user-images.githubusercontent.com/19729714/60426976-a89e1e80-9bed-11e9-98bd-a7e7f5f7b920.png)
![image](https://user-images.githubusercontent.com/19729714/60427029-c7041a00-9bed-11e9-86de-15a459fceb46.png)

**Comment by BreakBB** [comment]
To me it looks like you're training the model correctly.

So the issue is most likely you don't use enough examples to train your new "P_ORG" label and the "Apda distribution" entity.

[The docs](https://spacy.io/usage/training#ner) say:

> You should avoid iterating over the same few examples multiple times, or the model is likely to “forget” how to annotate other examples. If you iterate over the same few examples, you’re effectively changing the loss function. The optimizer will find a way to minimize the loss on your examples, without regard for the consequences on the examples it’s no longer paying attention to. One way to avoid this “catastrophic forgetting” problem is to “remind” the model of other examples by augmenting your annotations with sentences annotated with entities automatically recognized by the original model. Ultimately, this is an empirical process: you’ll need to experiment on your data to find a solution that works best for you.

and

> To keep the example short and simple, only a few sentences are provided as examples. In practice, you’ll need many more — a few hundred would be a good start. You will also likely need to mix in examples of other entity types, which might be obtained by running the entity recognizer over unlabelled sentences, and adding their annotations to the training set.

So you have to make sure to not only add examples with different entites and some without entities as well. Just experiment a bit and eventually your new model will improve.
